docs/low-code-spark/gems/machine-learning/ml-pinecone-lookup.md

```diff
@@ -0,0 +1,110 @@
+
+#### Properties
+
+Pinecone DB uses indexing to map the vectors to a data structure that will enable faster searching. The Pinecone Lookup Gem searches through a Pinecone index to identify embeddings with similarity to the input embedding. Enter the Pinecone **[(4) Index name](/docs/low-code-spark/gems/machine-learning/ml-pinecone-lookup.md#faq)** which you’d like to use for looking up embeddings.
+
+Select one of the Gem’s input columns with vector embeddings as the **(5) Vector column** to send to Pinecone’s API. The column [must](/docs/low-code-spark/gems/machine-learning/ml-pinecone-lookup.md#input-columns) be compatible with the Pinecone Index. To change the column’s datatype and properties, [configure](/docs/low-code-spark/gems/machine-learning/ml-pinecone-lookup.md#faq) the Gem(s) preceding the Pinecone Lookup Gem.
+
+Pinecone’s API can return multiple results. Depending on the use case, select the desired **(6) Number of results** sorted by similarity score. The result with highest similarity to the user’s text question will be listed first.
+
+### Input
+
+Pinecone Lookup requires a model_embedding column as input. Use one of Prophecy's Machine Learning Gems to provide the model_embedding. For example, the OpenAI Gem can precede the Pinecone Lookup Gem in the Pipeline. The OpenAI Gem, configured to `Compute a text embedding`, will output an openai_embedding column. This is a suitable input for the Pinecone Lookup Gem.
+
+| Column          | Description                                                                                                                                                                                                                                                                                                                                                                                                                                     | Required |
+| --------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |
+| model_embedding | array(float) - The format of this embedding is important. It must be an array of floating point numbers that matches the requirements of the Pinecone index. For example, we used a Pinecone index with `1536` dimensions, `Cosine` metric, and an `s1` pod type. So each record in the model_embedding column must be an array of `1536` floating point numbers between `-1 and 1`, such as `[-0.0018493991, -0.0059955865, ... -0.02498541]`. | True     |
```

Comment:
`floating point numbers between -1 and 1, such a` - is this accurate? 

This is definitely very common, but might not be accurate. You can have any ranged floats within the vectors (I believe).

Reply:
You're right - both OpenAI and Pinecone DB allow a wider range of values!
https://platform.openai.com/docs/guides/embeddings/what-are-embeddings
https://www.pinecone.io/learn/vector-search-basics/